primefaces/src/main/java/org/primefaces/util/ComponentUtils.java

```diff
@@ -607,4 +616,29 @@ public static String getDynamicColumnValueAsString(UIComponent component) {
 
         return table.getConvertedFieldValue(FacesContext.getCurrentInstance(), column);
     }
+
+    /**
+     * Encodes the given component locally as HTML.
+     * @param component The component to capture HTML output for.
+     * @param context The current FacesContext.
+     * @return The encoded HTML output of the given component.
+     * @throws UncheckedIOException Whenever something fails at I/O level. This would be quite unexpected as it happens locally.
+     */
+    public static String encodeComponent(UIComponent component, FacesContext context) {
+        ResponseWriter originalWriter = context.getResponseWriter();
+        FastStringWriter output = new FastStringWriter();
+        context.setResponseWriter(originalWriter.cloneWithWriter(output));
```

Comment:
@melloware this line throws NPE because originalWriter return nulls. I think that is why Omnifaces uses getRenderKit method to address that situations